### PR TITLE
release/downloadIstioCandidate.sh: check tarball url before download

### LINKED ISSUE #28582: downloadIstioCandidate shows success message when installing non-existent version
**Bug description**

Istio download reports success message when I download non-existent version:

```sh
samnaser@samnaser:~/workspace/istio-workspace/istio/release$ curl -L https://istio.io/downloadIstio | ISTIO_VERSION=1.28972897.8 TARGET_ARCH=x86_64 sh -
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   102  100   102    0     0    428      0 --:--:-- --:--:-- --:--:--   428
100  4277  100  4277    0     0  14209      0 --:--:-- --:--:-- --:--:-- 14209

Downloading istio-1.28972897.8 from https://github.com/istio/istio/releases/download/1.28972897.8/istio-1.28972897.8-linux.tar.gz ...tar (child): istio-1.28972897.8-linux.tar.gz: Cannot open: No such file or directory
tar (child): Error is not recoverable: exiting now
tar: Child returned status 2
tar: Error is not recoverable: exiting now
rm: cannot remove 'istio-1.28972897.8-linux.tar.gz': No such file or directory

Istio 1.28972897.8 Download Complete!

Istio has been successfully downloaded into the istio-1.28972897.8 folder on your system.

sh: line 122: cd: istio-1.28972897.8/bin: No such file or directory
Next Steps:
See https://istio.io/latest/docs/setup/install/ to add Istio to your Kubernetes cluster.

To configure the istioctl client tool for your workstation,
add the  directory to your environment path variable with:
         export PATH="$PATH:"

Begin the Istio pre-installation check by running:
         istioctl x precheck

Need more information? Visit https://istio.io/latest/docs/setup/install/
```

[ ] Docs
[x] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Extensions and Telemetry
[ ] Security
[ ] Test and Release
[x] User Experience
[ ] Developer Infrastructure

**Expected behavior**
Report that the install failed because specified version doesn't exist

### PR DESCRIPTION
**Pull request by daixiang0**
Signed-off-by: Xiang Dai <long0dai@foxmail.com>

Please provide a description for what this PR is for.

Fix #28582 

And to help us figure out who should review this PR, please 
put an X in all the areas that this PR affects.

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[X] User Experience
[ ] Developer Infrastructure

Pull Request Attributes

Please check any characteristics that apply to this pull request. 

[X] Does not have any changes that may affect Istio users.

### CODE REVIEW

File: release/downloadIstioCandidate.sh [hunk 1]
```diff
@@ -79,6 +79,10 @@ ARCH_URL="https://github.com/istio/istio/releases/download/${ISTIO_VERSION}/isti
 
 with_arch() {
   printf "\nDownloading %s from %s ...\n" "$NAME" "$ARCH_URL"
+  if ! curl --output /dev/null --silent --head --fail "$ARCH_URL"; then
+    printf "\n%s is not found, please specify right ISTIO_VERSION and TARGET_ARCH\n" "$ARCH_URL"
```
Comment (ericvn) [inline release/downloadIstioCandidate.sh:83]: Maybe change the message:
```suggestion
    printf "\n%s is not found, please specify a valid ISTIO_VERSION and TARGET_ARCH\n" "$ARCH_URL"
```

File: release/downloadIstioCandidate.sh [hunk 2]
```diff
@@ -87,6 +91,10 @@ with_arch() {
 
 without_arch() {
   printf "\nDownloading %s from %s ..." "$NAME" "$URL"
+  if ! curl --output /dev/null --silent --head --fail "$ARCH_URL"; then
+    printf "\n%s is not found, please specify right ISTIO_VERSION and TARGET_ARCH\n" "$ARCH_URL"
```
Comment (ericvn) [inline release/downloadIstioCandidate.sh:95]: same

**Comment by Monkeyanator** [comment]
small nit: could we do `set -e` on this script like we do with most of our bash scripts and then do something like [https://stackoverflow.com/a/19622569](https://stackoverflow.com/a/19622569) to give user-friendly error?

File: release/downloadIstioCandidate.sh [hunk 3]
```diff
@@ -87,6 +92,10 @@ with_arch() {
 
 without_arch() {
   printf "\nDownloading %s from %s ..." "$NAME" "$URL"
+  if ! curl --output /dev/null --silent --head --fail "$ARCH_URL"; then
+    printf "\n%s is not found, please specify a valid ISTIO_VERSION and TARGET_ARCH\n" "$ARCH_URL"
```
Comment (shamsher31) [inline release/downloadIstioCandidate.sh:96]: This should be `$URL` as it is without arch support.
```suggestion
    printf "\n%s is not found, please specify a valid ISTIO_VERSION and TARGET_ARCH\n" "$URL"
```

File: release/downloadIstioCandidate.sh [hunk 4]
```diff
@@ -79,6 +80,10 @@ ARCH_URL="https://github.com/istio/istio/releases/download/${ISTIO_VERSION}/isti
 
 with_arch() {
   printf "\nDownloading %s from %s ...\n" "$NAME" "$ARCH_URL"
+  if ! curl --output /dev/null --silent --head --fail "$ARCH_URL"; then
```
Comment (shamsher31) [inline release/downloadIstioCandidate.sh:83]: nit:
```suggestion
  if ! curl -o /dev/null -sIf "$ARCH_URL"; then
```

File: release/downloadIstioCandidate.sh [hunk 5]
```diff
@@ -87,6 +92,10 @@ with_arch() {
 
 without_arch() {
   printf "\nDownloading %s from %s ..." "$NAME" "$URL"
+  if ! curl --output /dev/null --silent --head --fail "$ARCH_URL"; then
```
Comment (shamsher31) [inline release/downloadIstioCandidate.sh:95]: nit:
```suggestion
  if ! curl -o /dev/null -sIf "$ARCH_URL"; then
```

**Review by shamsher31 — APPROVED** [review]
Thanks. LGTM.

File: release/downloadIstioCandidate.sh [hunk 6]
```diff
@@ -87,6 +92,10 @@ with_arch() {
 
 without_arch() {
   printf "\nDownloading %s from %s ..." "$NAME" "$URL"
+  if ! curl -o /dev/null -sIf "$URL"; then
+    printf "\n%s is not found, please specify a valid ISTIO_VERSION and TARGET_ARCH\n" "$URL"
+    exit
```
Comment (carolynhu) [inline release/downloadIstioCandidate.sh:97]: as this is without arch, we might need to replace `TARGET_ARCH` with `OS`?  or just print "xxx is not found, please specify a valid ISTIO_VERSION"

**Review by carolynhu — APPROVED** [review]

**Pull request merged**